datafusion/core/src/datasource/listing/url.rs

```diff
@@ -108,7 +109,8 @@ impl ListingTableUrl {
 
     /// Creates a new [`ListingTableUrl`] from a url and optional glob expression
     fn new(url: Url, glob: Option<Pattern>) -> Self {
-        let prefix = Path::parse(url.path()).expect("should be URL safe");
+        let decoded_path = percent_encoding::percent_decode_str(url.path()).decode_utf8_lossy();
```

Comment:
the `url.path()`, it is still percent encoded string.
`percent_decode_str` decodes it.

Reply:
I think this will cause the parse call below to fail in the event of non-ASCII characters.

I think it should just be a case of change `Path::parse` to `Path::from`